packages/client/hmi-client/src/App.vue

```diff
@@ -126,12 +129,12 @@ API.interceptors.response.use(
 	}
 );
 
+// Update the project when the projectId changes
 watch(
 	() => route.params.projectId,
 	async (projectId) => {
-		useProjects().get(projectId as IProject['id']);
-		// Refetch the list of all projects
-		useProjects().getAll();
+		await useProjects().get(projectId as IProject['id']);
+		await useProjects().getAll();
```

Comment:
If the project id is defined, why would we bother fetching all the projects in here? Isn't that slow and not needed? Why not defer that until it's necessary?

Reply:
Good point, there is no need to do this here.